app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.app.player.audio
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import javax.inject.Inject
+
+private const val TAG_DIALOG = "LANGUAGE_DIALOG"
+
+/** [ViewModel] for user app usage history. */
+@FragmentScope
+class AudioViewModel @Inject constructor(
+) : ViewModel() {
+
+  var currentlanguageCode: String = "en"
+
+  val audioLanguageCode: String? by lazy {
+    getCurrentLanguageCode()
+  }
+
+  fun setAudioLanguageCode(languageCode: String) {
+    currentlanguageCode = languageCode
+  }
+
+  private fun getCurrentLanguageCode(): String {
+    return currentlanguageCode
+  }
+
+  fun languageSelectionClicked(fragmentManager: FragmentManager) {
+    val fragmentTransaction = fragmentManager.beginTransaction()
+    val prev = fragmentManager.findFragmentByTag(TAG_DIALOG)
+    if (prev != null) {
+      fragmentTransaction.remove(prev)
+    }
+    fragmentTransaction.addToBackStack(null)
```

Comment:
Rather than relying on the backstack here, could we just retrieve the existing dialog via findFragmentByTag() and dismiss it, then show a new dialog?

Reply:
I have updated this code as per [this](https://stackoverflow.com/questions/22474584/remove-old-fragment-from-fragment-manager) resource but it almost looks like previous code, so not sure if it is correct.

Reply:
See my later thread. I was hoping we could use dismiss() on the dialog fragment, but your solution is more correct so we'll keep this.